src/tests/UnitTests/EnsureEnumParamTests.cs

```diff
@@ -9,29 +9,121 @@ public class EnsureEnumParamTests : UnitTestBase
+                string.Format(ExceptionMessages.Enum_IsValidEnum, item, typeof(EnumDummy)),
+                () => Ensure.Enum.IsStrictlyDefined(item, ParamName),
+                () => EnsureArg.EnumIsStrictlyDefined(item, ParamName),
+                () => Ensure.That(item, ParamName).IsStrictlyDefined());
+        }
+
+        [Theory]
+        [InlineData(FlagDummy.Bar)]
+        [InlineData(FlagDummy.Bar | FlagDummy.Baz)]
+        public void FlagIsDefined_ShouldNotThrow(object flagsAsObject)
+        {
+            var item = (FlagDummy)flagsAsObject;
+
+            ShouldNotThrow(
                 () => Ensure.Enum.IsDefined(item, ParamName),
                 () => EnsureArg.EnumIsDefined(item, ParamName),
                 () => Ensure.That(item, ParamName).IsDefined());
         }
 
-        private enum Only1IsValid
+        [Theory]
+        [InlineData((FlagDummy)3)]
+        [InlineData((FlagDummy)0)]
+        public void FlagNotDefined_ShouldThrow(object itemAsObject)
+        {
+            var item = (FlagDummy)itemAsObject;
+            ShouldThrow<ArgumentOutOfRangeException>(
+                string.Format(ExceptionMessages.Enum_IsValidEnum, item, typeof(FlagDummy)),
+                () => Ensure.Enum.IsDefined(item, ParamName),
+                () => EnsureArg.EnumIsDefined(item, ParamName),
+                () => Ensure.That(item, ParamName).IsDefined());
+        }
+
+        [Fact]
+        public void FlagIsDefined_Strictly_ShouldNotThrow_IfNotCombined()
+        {
+            var item = FlagDummy.Bar;
+
+            ShouldNotThrow(
+                () => Ensure.Enum.IsStrictlyDefined(item, ParamName),
+                () => EnsureArg.EnumIsStrictlyDefined(item, ParamName),
+                () => Ensure.That(item, ParamName).IsStrictlyDefined());
+        }
+
+        [Fact]
+        public void FlagIsDefined_Strictly_ShouldThrow_IfCombined()
+        {
+            var item = FlagDummy.Bar | FlagDummy.Baz;
+
+            ShouldThrow<ArgumentOutOfRangeException>(
+                string.Format(ExceptionMessages.Enum_IsValidEnum, item, typeof(FlagDummy)),
+                () => Ensure.Enum.IsStrictlyDefined(item, ParamName),
+                () => EnsureArg.EnumIsStrictlyDefined(item, ParamName),
+                () => Ensure.That(item, ParamName).IsStrictlyDefined());
+        }
+
+        [Fact]
+        public void FlagNotDefined_Strictly_ShouldThrow()
+        {
+            var item = (FlagDummy)3;
+            ShouldThrow<ArgumentOutOfRangeException>(
+                string.Format(ExceptionMessages.Enum_IsValidEnum, item, typeof(FlagDummy)),
+                () => Ensure.Enum.IsStrictlyDefined(item, ParamName),
+                () => EnsureArg.EnumIsStrictlyDefined(item, ParamName),
+                () => Ensure.That(item, ParamName).IsStrictlyDefined());
+        }
+
+        private enum EnumDummy : byte
         {
             Valid = 1
         }
+
+        [Flags]
+        private enum FlagDummy : byte
```

Comment:
[nit] I'd avoid Dummy in tests myself usually. The type is a type, just because you're using it for tests doesn't make it not a type of some sort anymore :-P I'd name it for what it is something like `TestFlagsEnum` myself

Reply:
Renamed it. Also added yours `Mine` flags as `TestFlagsOfWhateverPower`. 😉